src/js/graphql_endpoint/modules/root_query.ts

```diff
@@ -199,84 +208,98 @@ const handleLensScope = async (parent: MysteryParentType, args: LensArgs) => {
+  readonly offset: number;
 }
 
 interface LensArgs {
-  lens_name: string;
+  readonly lens_name: string;
 }
 
-const RootQuery = new GraphQLObjectType({
-  name: "RootQueryType",
-  fields: {
-    lenses: {
-      type: GraphQLList(LensNodeType),
-      args: {
-        first: {
-          type: new GraphQLNonNull(GraphQLInt),
+function getRootQuery(): GraphQLObjectType {
+  return new GraphQLObjectType({
+    name: "RootQueryType",
+    fields: {
+      lenses: {
+        type: GraphQLList(LensNodeType),
+        args: {
+          first: {
+            type: new GraphQLNonNull(GraphQLInt),
+          },
+          offset: {
+            type: new GraphQLNonNull(GraphQLInt),
+          },
         },
-        offset: {
-          type: new GraphQLNonNull(GraphQLInt),
+        resolve: async (parent: MysteryParentType, args: RootQueryArgs) => {
+          console.debug("lenses query arguments", args);
+          const first = args.first;
+          const offset = args.offset;
+          // #TODO: Make sure to validate that 'first' is under a specific limit, maybe 1000
+          console.debug("Making getLensesQuery");
+          const lenses = await getLenses(getDgraphClient(), first, offset);
+          console.debug(
+            "returning data from getLenses for lenses resolver",
+            lenses
+          );
+          return lenses;
         },
       },
-      resolve: async (parent: MysteryParentType, args: RootQueryArgs) => {
-        console.debug("lenses query arguments", args);
-        const first = args.first;
-        const offset = args.offset;
-        // #TODO: Make sure to validate that 'first' is under a specific limit, maybe 1000
-        console.debug("Making getLensesQuery");
-        const lenses = await getLenses(getDgraphClient(), first, offset);
-        console.debug(
-          "returning data from getLenses for lenses resolver",
-          lenses
-        );
-        return lenses;
-      },
-    },
-    lens_scope: {
-      type: LensNodeType,
-      args: {
-        lens_name: { type: new GraphQLNonNull(GraphQLString) },
-      },
-      resolve: async (parent: MysteryParentType, args: LensArgs) => {
-        try {
-          console.debug("lens_scope args: ", args);
-          let response = await handleLensScope(parent, args);
-          console.debug("lens_scope response: ", response);
-          return response;
-        } catch (e) {
-          console.error("Error in handleLensScope: ", e);
-          throw e;
-        }
+      lens_scope: {
+        type: LensNodeType,
+        args: {
+          lens_name: { type: new GraphQLNonNull(GraphQLString) },
+        },
+        resolve: async (parent: MysteryParentType, args: LensArgs) => {
+          try {
+            console.debug("lens_scope args: ", args);
+            let response = await handleLensScope(parent, args);
+            console.debug("lens_scope response: ", response);
+            return response;
+          } catch (e) {
```

Comment:
all of this hasn't changed, besides some formatting and tabbing it in (because it's now a function)